core/src/main/scala/Config.scala

```diff
@@ -686,18 +767,24 @@ object Config {
       monitoringPort = cfg.require[Int]("monitoring-port")
     )
 
-  private def readGithub(cfg: KConfig): GithubConfig =
-    GithubConfig(
-      domain = cfg.lookup[String]("domain"),
-      clientId = cfg.require[String]("client-id"),
-      clientSecret = cfg.require[String]("client-secret"),
-      redirectUri = cfg.require[String]("redirect-uri"),
-      scope = cfg.require[String]("scope"),
-      systemAccessToken = AccessToken(cfg.require[String]("access-token")),
-      systemUsername = cfg.require[String]("system-username"),
-      organizationBlacklist = cfg.lookup[List[String]]("organization-blacklist").getOrElse(Nil),
-      organizationAdminList = cfg.lookup[List[String]]("organization-admins").getOrElse(Nil)
+  private def readGithub(cfg: KConfig): ScmConfig = {
+    val service = cfg.lookup[String]("service").getOrElse("github")
```

Comment:
The service configuration param is used to choose the scm interpreter: Gitlab or GIthub (default).

Reply:
For reading configuration for different interpreters I might do something like
```scala
kfg.lookup[String]("scm") match {
  case Some("github") => readGithub(kfg.subconfig("github"))
  case Some("gitlab") => readGitlab(kfg.subconfig("gitlab"))
  case _             => None
}```

Reply:
For the time being is the same configuration.